src/Elmish.WPF/ViewModel.fs

```diff
@@ -49,7 +50,8 @@ and [<AllowNullLiteral>] ViewModel<'model, 'msg>
       ( initialModel: 'model,
         dispatch: 'msg -> unit,
         bindingSpecs: BindingSpec<'model, 'msg> list,
-        config: ElmConfig )
+        config: ElmConfig,
+        threadDispatcher: Dispatcher)
```

Comment:
Why is a `Dispatcher` injected here? Is it to make `ViewModel` testable? If so, since we only seem to need the `Invoke` method fro the dispatcher, perhaps it could be made even simpler and more testable by changing the type from `Dispatcher` to `unit -> unit`. What do you think? (I have no idea how simple `Dispatcher` is to create/mock, but I like to keep things as simple as reasonably possible.)

Also, do you think it would be clearer if it was given a name that indicates what it dispatches to? (Such as `uiDispatcher`, `mainDispatcher`, or any better suggestion you might have.)

Reply:
This one is if we ever open the library to writing plug-ins for extensible application. I found out while trying to write an AutoCAD palette that `Application.Current` return `null` in such a context. Passing the thread dispatcher from the control to bind solve the problem, and hopefully will in other contexts.

Anyway you're right about passing the function instead the `Dispatcher` object, i'll make this change and call it `uiDispatcher`.

Reply:
This one has been a little tougher than expected since `Dispatcher.Invoke` demands an `Action` delegate. I ended up with a `(unit -> unit) -> unit` parameter.

Reply:
Of course, my bad. `(unit -> unit) -> unit` is correct.